Fix comparisons of const char* agains '\0' where obviously
the first character of the array was meant to be checked.

--- prog/check_funs.cpp	2011-07-04 11:17:27.000000000 +0200
+++ prog/check_funs.cpp	2017-07-23 17:37:57.880903979 +0200
@@ -647,7 +647,7 @@
     }
   }
   if (i == width-1) {
-    if (word == '\0')
+    if (*word == '\0')
       put(out,' ');
     else if (word[len] == '\0')
       put(out, word, len);
--- modules/filter/tex.cpp	2011-07-02 23:09:09.000000000 +0200
+++ modules/filter/tex.cpp	2017-07-23 17:40:05.089075293 +0200
@@ -174,7 +174,7 @@
 
     if (c == '{') {
 
-      if (top.in_what == Parm || top.in_what == Opt || top.do_check == '\0')
+      if (top.in_what == Parm || top.in_what == Opt || *top.do_check == '\0')
 	push_command(Parm);
 
       top.in_what = Parm;
